build_linux_nvidia.sh

```diff
@@ -0,0 +1,381 @@
+      echo 'export CUDNN_DIR="/usr/include/'${ARCH}'-linux-gnu/"' >> ${HOME}/.bashrc
+      echo 'export LD_LIBRARY_PATH="/usr/lib/'${ARCH}'-linux-gnu/:$LD_LIBRARY_PATH"' >> ${HOME}/.bashrc
+      source ${HOME}/.bashrc
+      echo_green "Please re-run this script for changes to apply!"
+      exec bash
+    else
+      echo_red "Please Install TensorRT, CUDNN and add TENSORRT_DIR, CUDNN_DIR to environment variables before running this script!"
+      exit
+    fi
+  fi
+
+  # Determine TensorRT version and set paths accordingly
+  echo "Checking TensorRT version...Please verify the detected versions below:"
+  if [[ "$ARCH" == aarch64 ]]; then
+    cat /usr/include/${ARCH}-linux-gnu/NvInferVersion.h | grep NV_TENSORRT
+  else
+    cat ${TENSORRT_DIR}/include/NvInferVersion.h | grep NV_TENSORRT
+  fi
+  read -p "Is TensorRT >=8.0.1.6 installed? (Always installed on Jetson) (y/n)" -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Nn]$ ]]
+  then
+      echo "Error: You must install TensorRT before installing MMDeploy!"
+      exit
+  fi
+
+  # opencv
+  sudo apt-get install libopencv-dev
+}
+
+py_venv() {
+  # deactivate venv, if it has already been activated
+  deactivate
+
+  #check for python installed version
+  pyv="$(python3 -V 2>&1)"
+  pyv_old="Python 3.6"
+
+  if echo "$pyv" | grep -q "$pyv_old"; then
+      # use python 3.6
+      curl https://bootstrap.pypa.io/pip/3.6/get-pip.py -o get-pip.py
+  else
+      # use python >=3.7
+      curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+  fi
+
+  # dependencies
+  sudo apt-get install protobuf-compiler libprotoc-dev libssl-dev curl ninja-build -y
+  sudo apt-get install libopenblas-dev python3-venv python3-dev python3-setuptools -y
+  sudo python3 get-pip.py
+  pip3 install testresources
+  pip3 install --upgrade setuptools wheel
+
+  if [ -d "${PYTHON_VENV_DIR}" ]; then
+    read -p "Reinstall existing Python venv ${PYTHON_VENV_DIR}? (y/n)" -n 1 -r
+    echo
+    if [[ $REPLY =~ ^[Yy]$ ]]
+    then
+      rm -r ${PYTHON_VENV_DIR}
+      python3 -m venv ${PYTHON_VENV_DIR} --system-site-packages #system site packages to keep trt from system installation
+    fi
+  else
+    python3 -m venv ${PYTHON_VENV_DIR} --system-site-packages #system site packages to keep trt from system installation
+  fi
+
+  source ${PYTHON_VENV_DIR}/bin/activate
+  python3 get-pip.py
+  pip3 install testresources
+  pip3 install --upgrade setuptools wheel
+
+  # Latest PIL is not compatible with mmcv=1.4.1
+  pip install Pillow==7.0.0
+
+  if [[ "$ARCH" == aarch64 ]]
+  then
+    # protofbuf on jetson is quite old - must be upgraded
+    pip3 install --upgrade protobuf
+    # Install numpy 1.19.4 as newer versions might give "Illegal instruction (core dumped)" on Jetson
+    # TODO Numpy might be installed per default so we should not remove it.
+    pip3 install numpy==1.19.4
+  else
+    pip3 install numpy
+  fi
+  pip3 install opencv-python
+  pip3 install matplotlib
+
+  # pytorch, torchvision, torchaudio
```

Comment:
Can we make pytorch optional?

Reply:
@grimoire Sure, but wont it be required for MMDeploy?

Reply:
We hope users can export the ONNX file on the server and convert it to TensorRT on the board (in the future). Less dependency means less storage usage and faster installation.

Reply:
@grimoire That makes sense. That functionality would also be very valuable for us. For now, we generate the models on the Jetson and copy them to a NAS.